sdk/producer.go

```
@@ -0,0 +1,939 @@
+func (p *Producer) nextPartition() int {
+	idx := int((atomic.AddUint32(&p.rr, 1) - 1) % uint32(p.partitions))
+	return idx
+}
+
+func (p *Producer) CreateTopic(topic string, partitions int) error {
+	brokerAddr := p.config.BrokerAddrs[0]
+	conn, err := net.Dial("tcp", brokerAddr)
+	if err != nil {
+		return fmt.Errorf("connect: %w", err)
+	}
+	defer conn.Close()
```

golangci-lint (2.11.4)

[error] Error return value of `conn.Close` is not checked

(errcheck)